### PR TITLE
Adding capnp-support

### PR DESCRIPTION
**Pull request by karlnorling**
- Building docker
- Schema version 2 is using Cap'n Proto
- Schema version 1 backwards compatibility
- See: https://github.com/capnproto/node-capnp
- Tests
- Coverage reports
- Github workflow actions to run test

### CODE REVIEW

**Comment by karlnorling** [comment]
Let's make sure everything is 💯 before merging this.

File: root/usr/bin/redsift/run-es6.js [hunk 1]
```diff
@@ -22,6 +24,33 @@ const DRY = init.DRY;
...
+const readFileContent = (file) => {
+  var ipc_root = process.env.IPC_ROOT;
+  var filename = ipc_root + '/' + file;
+  return FS.readFileSync(filename);
+}
+
+const detectAndParseCapnProto = (req) => {
+  const KERASH_ID_KEY = 'rstid:kerash';
+  if (req.in.data && sift['schema-version'] === 2) {
+    console.log('Cap’n Proto Schema detected');
```
Comment (rsdy) [inline root/usr/bin/redsift/run-es6.js]: Probably wanna change this to `console.debug`

**Comment by deepakprabhakara** [comment]
Schema version 2 also needs to read the value from file.

**Comment by karlnorling** [comment]
Adding some tests.

**Comment by karlnorling** [comment]
@cv711 @rsdy @deepakprabhakara this is good to go!

File: root/usr/bin/redsift/protocol.js [hunk 2]
```diff
@@ -1,5 +1,14 @@
 'use strict';
 
+const convert = require('./convert.js');
+const fs = require('fs');
+
+const readFileContent = (file) => {
+  const ipc_root = process.env.IPC_ROOT;
+  const filename = ipc_root + '/' + file;
```
Comment (cv711) [inline root/usr/bin/redsift/protocol.js]: better to use `path.join` here

File: root/usr/bin/redsift/run-es6.js [hunk 3]
```diff
@@ -53,21 +66,35 @@ nodes.forEach(function (i) {
   reply.rcvmaxsize(-1);
   reply.connect('ipc://' + path.join(IPC_ROOT, i + '.sock'));
   reply.on('data', function (msg) {
+    let req = JSON.parse(msg);
```
Comment (cv711) [inline root/usr/bin/redsift/run-es6.js]: this should be one line lower to get a correct measurement for the stats

File: root/usr/bin/redsift/protocol.js [hunk 4]
```diff
@@ -10,60 +19,91 @@ function flattenNestedArrays(value) {
...
   // if node() returns a Promise.all([...]), remove the nesting
-  var flat = flattenNestedArrays(value);
+  let flat = flattenNestedArrays(value);
   //console.log('REP-FLAT:', flat);
   flat.forEach(function (i) {
-    i = b64Encode(i);
+    i = convert.b64Encode(i);
   });
-  var encodeTime = process.hrtime(startEncode);
+  let encodeTime = process.hrtime(startEncode);
```
Comment (cv711) [inline root/usr/bin/redsift/protocol.js]: all `let` vars here can be `const`s

File: root/usr/bin/redsift/protocol.js (hunk 2 above)
Comment (cv711) [inline root/usr/bin/redsift/protocol.js]: better to use `path.join` here

**Review by cv711 — CHANGES_REQUESTED** [review]
some small JS changes

File: root/usr/bin/redsift/protocol.js (hunk 2 above)
Comment (karlnorling) [reply root/usr/bin/redsift/protocol.js]: `${icp_root} / ${file}` ?

Comment (karlnorling) [reply root/usr/bin/redsift/protocol.js]:
```
const readFileContent = (file) =>
  fs.readFileSync(`${process.env.IPC_ROOT}/${file}`);
```

**Comment by karlnorling** [comment]
@cv711 pushed code again! _(sorry did a git rebase -i to squash the commits)_

File: root/usr/bin/redsift/protocol.js (hunk 2 above)
Comment (cv711) [reply root/usr/bin/redsift/protocol.js]: I meant `filename = path.join(ipc_root, file)`

**Review by cv711 — APPROVED** [review]
LGTM

**Review by cv711 — APPROVED** [review]
LGTM#2

**Pull request merged**